test/extended/marketplace/marketplace_multi.go

```diff
@@ -0,0 +1,163 @@
+package marketplace
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[Feature:Marketplace] Marketplace support multi format operators", func() {
+
+	defer g.GinkgoRecover()
+
+	var (
+		oc            = exutil.NewCLI("marketplace", exutil.KubeConfigPath())
+		marketplaceNs = "openshift-marketplace"
+		resourceWait  = 60 * time.Second
+
+		opsrcYamltem = exutil.FixturePath("testdata", "marketplace", "opsrc", "02-opsrc.yaml")
+	)
+
+	g.AfterEach(func() {
+		//clear the sub,csv resource
+		allresourcelist := [][]string{
+			{"operatorsource", "flatted", marketplaceNs},
+			{"operatorsource", "nested", marketplaceNs},
+			{"operatorsource", "multiformat", marketplaceNs},
```

Comment:
Would it be sufficient to only test with a multi-format Appregistry namespace? That way, if we updated the `marketplace_e2e` appr namespace to be multi-format, we could avoid creating new namespaces.

Reply:
Hmm the original story and test case is : support only flattented , only nested, nested+ flattented, 3 kinds of opsrc . If @kevinrizza  say it is ok , then we need to push one "nested" operator and one "with error" operator to `marketplace_e2e` appr namespace to be multi-format and "with error operator" . Then we could avoid creating new ns.

Reply:
I think it would be fine to just test the multi-format registry namespace. But I think it should not be the `marketplace_e2e` test namespace because that should, ideally, only be updated for the e2e tests in the marketplace repository. Otherwise someone could inadvertently break a test in a repo while trying to write a new test for the other repo. Can we just use this `multiformat` namespace for these tests?